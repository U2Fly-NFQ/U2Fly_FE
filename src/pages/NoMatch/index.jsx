import { Button, Result } from 'antd'
import React from 'react'
import { useNavigate } from 'react-router-dom'
import notFoundImg from '../../assets/images/system/not-found.png'

const NoMatch = () => {
  const navigate = useNavigate()

  return (
    <>
      <Result
        style={{
          paddingTop: '128px',
        }}
        icon={
          <img
            src={notFoundImg}
            style={{
              maxWidth: '100%',
            }}
          />
        }
        title="404"
        subTitle="Sorry, the page you visited does not exist."
        extra={
          <Button type="primary" onClick={() => navigate('/')}>
            Back Home
          </Button>
        }
      />
    </>
  )
}

export default NoMatch
