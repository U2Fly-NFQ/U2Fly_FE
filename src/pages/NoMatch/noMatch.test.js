import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import NoMatch from './'
import { fireEvent, render, screen } from '@testing-library/react'
describe('No match image test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <NoMatch />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should onClick Correctly', async () => {
    render(
      <Router>
        <NoMatch />
      </Router>
    )
    const backToHome = screen.getByText('Back Home')
    fireEvent.click(backToHome)
  })
})
