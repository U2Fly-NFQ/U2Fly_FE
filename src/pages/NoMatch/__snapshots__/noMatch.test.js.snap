// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No match image test renders correctly 1`] = `
<div
  className="ant-result ant-result-info"
  style={
    Object {
      "paddingTop": "128px",
    }
  }
>
  <div
    className="ant-result-icon"
  >
    <img
      src="not-found.png"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    />
  </div>
  <div
    className="ant-result-title"
  >
    404
  </div>
  <div
    className="ant-result-subtitle"
  >
    Sorry, the page you visited does not exist.
  </div>
  <div
    className="ant-result-extra"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Back Home
      </span>
    </button>
  </div>
</div>
`;
